Java/core/src/main/java/com/amazon/randomcutforest/tree/NodeView.java

```diff
@@ -76,6 +76,12 @@ public double probailityOfSeparation(float[] point) {
         return nodeStore.probabilityOfCut(currentNodeOffset, point, pointStoreView, currentBox);
     }
 
+    @Override
+    public int getLeafPointIndex() {
+        checkArgument(isLeaf(), "incorrect call");
```

Comment:
use `checkState` instead of `checkArgument` and give a better error message like "cannot invoke 'getLeafPointIndex' from a non-leaf node".

Reply:
Done.